src/template-cache.js

```diff
@@ -5,22 +5,27 @@ import _               from 'underscore';
 import Backbone        from 'backbone';
 import MarionetteError from './error';
 
-// Manage templates stored in `<script>` blocks,
-// caching them for faster access.
-var TemplateCache = function(templateId) {
-  this.templateId = templateId;
-};
-
-// TemplateCache object-level methods. Manage the template
-// caches from these method calls instead of creating
-// your own TemplateCache instances
-_.extend(TemplateCache, {
-  templateCaches: {},
+class TemplateCache {
+  constructor(templateId) {
+    this.templateId = templateId;
+  }
+
+  static get templateCaches() {
+    if (!this._templateCaches) {
+      this._templateCaches = {};
+    }
+
+    return this._templateCaches;
+  }
+
+  static set templateCaches(cache) {
+    this._templateCaches = cache;
```

Comment:
i feel like since we are using this is the body static might be the wrong type here